src/main/java/mate/academy/dao/BookDaoImpl.java

```
@@ -0,0 +1,116 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+
+            int i = preparedStatement.executeUpdate();
+            if (i < 1) {
+                throw new DataProcessingException("Can`t create book" + book, null);
+            }
+
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+                return book;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create book" + book, e);
+        }
+        return null;

```

Issue: The `create` method returns `null` if the generated keys are not present. According to the checklist, methods should not return `null` for failed operations but should throw an exception instead. Please throw a `DataProcessingException` here to comply with best practices and checklist requirements.